client/src/components/category-list-item/category-list-item.js

```diff
@@ -0,0 +1,37 @@
+import Component from '@/components/Component';
+import ImgBox from '@/components/img-box/img-box';
+
+import styles from '@/styles/components/category-list-item/category-list-item.module.scss';
+
+// props : { type, text }
+export default class CategoryListItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.$dom = this.createDom('div', {
+      className: styles['category-list-item'],
+    });
+
+    this.ImgBox = new ImgBox({
+      type: 'small',
+      imageURL: this._props.imageURL,
+      onClick: () => {},
+    });
+
+    this.render();
+    this.addEvent();
+  }
+
+  render = () => {
+    this.$dom.innerHTML = `
+      <div class="${styles['ImgBox']}"></div>
+      <div class="${styles['title']}">${this._props.title}</div>
+    `;
+
+    this.replaceElement(this.$dom.querySelector(`.${styles['ImgBox']}`), this.ImgBox.$dom);
+  };
+
+  addEvent = () => {
+    this.$dom.addEventListener('click', this._props.onClick);
```

Comment:
```suggestion
    if (this._props.onClick) {
      this.$dom.addEventListener('click', this._props.onClick);
    }
```

아래에 있는 코드를 보면서 느낀건데
```javascript
this.ImgBox = new ImgBox({
  type: 'small',
  imageURL: this._props.imageURL,
  onClick: () => {},
});
```

이 컴포넌트에서는 ImgBox는 이미지 UI의 역할을 할 뿐이지 click 이벤트 처리는 하지 않고 있음에도 불과하고 onClick props를 넘겨주고 있습니다.

제안한 것과 같이 onClick이 있으면 event listener를 등록하고 그렇지 않으면 넘어가도록 코드를 바꿨으면 좋겠습니다.

Reply:
저도 이 부분이 조금 걸렸는데 위와 같이 하면 좋을 것 같습니다. 반영하겠습니다.